spec/models/embedded_ansible_worker/runner_spec.rb

```diff
@@ -10,7 +10,9 @@
     let(:runner) {
       worker
       allow_any_instance_of(described_class).to receive(:worker_initialization)
-      described_class.new(:guid => worker_guid)
+      r = described_class.new(:guid => worker_guid)
+      allow(r).to receive(:worker).and_return(worker)
```

Comment:
I think the answer here will be to just remove the zone/role stuff from settings. Then we don't have to deal with multiple sources of information for this type of thing.

Also, I don't really want to do that in this PR. I'll work on that as a follow up and fix this up there.